### PR TITLE
Clean up worker inputs API

### PR DESCRIPTION
**Pull request by Toxantron**
With the recent additions to the WorkerSupport API in MORYX-Factory 6.1 it became clear to give the API an overhaul for MORYX 8.

### CODE REVIEW

File: src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs [hunk 1]
```diff
@@ -14,34 +15,22 @@ public interface IVisualInstructor : IResource
...
         /// Have to be cleared with the <see cref="Clear"/> method
         /// </summary>
-        long Display(string sender, VisualInstruction[] instructions);
+        long Display(ActiveInstruction instruction);
 
         /// <summary>
         /// Only display these instructions
         /// Instruction will automatically cleared after the given time
         /// </summary>
-        void Display(string sender, VisualInstruction[] instructions, int autoClearMs);
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs:23]: When you are on it, can you proof read the comments once 😅 
In this case "Instruction will be cleared automatically after the given <param>...."

File: src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs [hunk 2]
```diff
@@ -14,34 +15,22 @@ public interface IVisualInstructor : IResource
...
+        void Display(ActiveInstruction instruction, int autoClearMs);
 
         /// <summary>
-        /// Execute these instructions based on the given activity and report the result on completion
-        /// Can (but must not) be cleared with the <see cref="Clear"/> method
+        /// Execute these instructions based on the given activity and report the result and inputs on completion
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs:27]: From an outside perspective: This method signature says nothing about an activity, also the phrase "report the results and inputs" will probably be forgotten to be changed when the ActiveInstructionResult changes in the future (and then be incomplete or even misleading ).

File: src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0
 
 using Moryx.AbstractionLayer.Resources;
+using System;
 
 namespace Moryx.ControlSystem.VisualInstructions
 {
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs:8]: Based o the discussion in #37 would it be to much to ask of you whether you could add the params and return xml documentation for the methods in here? What is for example the return value of the display method? 😅
And while you are at it, the explenation you gave in the same thread on what the IVisualInstructor and what the IWorkerSupport Facade are good for, do you think you could put that somehow into the interface xml comments? 🤔

File: src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs [hunk 4]
```diff
@@ -14,34 +15,22 @@ public interface IVisualInstructor : IResource
         /// Only display these instructions
         /// Have to be cleared with the <see cref="Clear"/> method
         /// </summary>
-        long Display(string sender, VisualInstruction[] instructions);
```
[the remark this answers is not in the document]
Comment (Toxantron) [reply src/Moryx.ControlSystem/VisualInstructions/IVisualInstructor.cs:17]: The `ActiveInstruction` calls it `Title`. But I will update the argument in the extensions.

File: src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs [hunk 5]
```diff
@@ -72,10 +72,10 @@ private enum TestResults4
         public void GetAllValuesIfThereAreHiddenAndNotDecoratedResults()
         {
             // Act
-            var instructionResult = new EnumInstructionResult(typeof(TestResults4), result => { });
+            var instructionResult = EnumInstructionResult.PossibleResults(typeof(TestResults4));
 
             // Assert
-            Assert.AreEqual(2, instructionResult.Results.Count(), "There should be no results because there are hidden and not decorated results");
+            Assert.AreEqual(2, instructionResult.Count, "There should be no results because there are hidden and not decorated results");
```
[the remark this answers is not in the document]
Comment (Toxantron) [reply src/Tests/Moryx.ControlSystem.Tests/EnumInstructionResultTests.cs:78]: It seems the behavior is wrong and someone just adjusted the test. I will investigate and fix this.

**Pull request merged**